### PR TITLE
[ledger-api] - metadata for invalid deduplication period [KVL-1170]

### PR DESCRIPTION
**Pull request by nicu-da**
CHANGELOG_BEGIN
[ledger-api] - Return max_deduplication_duration as part of the metadata sent with the gRPC status for commands rejected because of invalid deduplication periods
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/FieldValidations.scala [hunk 1]
```diff
@@ -162,6 +162,7 @@ class FieldValidations private (errorFactories: ErrorFactories) {
               fieldName,
               exceedsMaxDurationMessage,
               definiteAnswer = Some(false),
+              maxDeduplicationTime,
```
Comment (fabiotudone-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/FieldValidations.scala:165]: Could you rename it to `maxDeduplicationDuration`?

**Review by fabiotudone-da — COMMENTED** [review]
Is the KV part coming in a distinct PR?

**Review by hubert-da — APPROVED** [review]

**Comment by nicu-da** [comment]
> Is the KV part coming in a distinct PR?

I think I'm missing the context where we would have a specific kv occurrence for this. This only happens during the submissions. Can you provide any context I'm missing to figure this out?

**Review by SamirTalwar-DA — APPROVED** [review]

**Comment by fabiotudone-da** [comment]
> > Is the KV part coming in a distinct PR?
> 
> I think I'm missing the context where we would have a specific kv occurrence for this. This only happens during the submissions. Can you provide any context I'm missing to figure this out?

Oh we never reject with this reason on the committer, always in the participant, right?

**Comment by nicu-da** [comment]
> > > Is the KV part coming in a distinct PR?
> > 
> > 
> > I think I'm missing the context where we would have a specific kv occurrence for this. This only happens during the submissions. Can you provide any context I'm missing to figure this out?
> 
> Oh we never reject with this reason in the committer, always in the participant, right?

That is correct, this never reaches the committer.

**Review by fabiotudone-da — APPROVED** [review]

**Pull request merged**